packages/database/src/models/SurveyScreenComponent.ts

```diff
@@ -105,10 +105,11 @@ export class SurveyScreenComponent extends Model {
   }
 
   forResponse() {
-    const { options, ...values } = this.dataValues;
+    const { options, dataElement, ...values } = this.dataValues;
     return {
       ...values,
       options: safeJsonParse(options),
+      dataElement: dataElement ? dataElement.forResponse() : dataElement,
```

Comment:
I had to do this because when returning the dataElement in the survey response endpoint with zod it was interpreting the dataElement objects as strings instead of objects. I think this is generally better since it's more explicit.